Automattic/wp-calypso
# Stepper: incorporate setIntentOnSite

## Changes proposed in this Pull Request

Currently in `/start` the intent is saved at `/sites/$site_id/site-intents` after the user completes the sign-up flow. If they leave prematurely this is not saved. This PR incorporates the same API call when the user exits the flow through one of the available step exits.

I added a new prop to the `useStepNavigation` hook called `exit` to handle all the redirects away from the flow. This comes with an added callback where I passed the save intent action.

## Testing instructions

1. Pull branch and `yarn start`
2. Go to... `http://calypso.localhost:3000/setup/intent?siteSlug=` [ YOUR SITE HERE ]
3. Open the API console - https://developer.wordpress.com/docs/api/console-2014/ ( make sure you set it to `wp rest API` and `wpcom/v2` )
4. Click through the options and check `/sites/YOUR-SITE-ID/site-intent/`
5. You should not see the site intent populate or update until you click through the sign-up flow.

Fixes #62725
